### ISSUE TITLE
Redox doesn't start on real hardware 

### ISSUE THREAD

**Issue by Pharaone** [redox-os/redox#1019]
**Reproduction**: [burn the redox's  iso into an USB drive and boot it from a pc.]

**Expected behavior**: [the OS starts.]

**Actual behavior**: [all crashes during the boot, so the pc restarts and restarts endlessy.

**Build information**: [Acer aspire one d225,Intel atom n450, 2,00GB of ram.]
**Redox release**: [0.3.0]

**Comment by Pharaone** [comment]
![index](https://user-images.githubusercontent.com/17162630/29037067-51a9501c-7ba2-11e7-8461-57d205e5cd61.jpg)

**Comment by ids1024** [comment]
I don't see an error of any kind in that screenshot...

How much RAM does the computer you're testing on have?

**Comment by Pharaone** [comment]
2gb

Il 07 ago 2017 7:10 PM, "Ian Douglas Scott" <notifications@github.com> ha
scritto:

**Comment by jackpot51** [comment]
Yeah, you will need a more RAM to load it. The live ISO has to store the whole uncompressed filesystem in RAM

**Comment by Pharaone** [comment]
Will it change with the next versions?

**Comment by ids1024** [comment]
@jackpot51 I guess it's not too straightforward to make it smarter, and require only the RAM for currently used space, right? I think that would require that Redox have something like Linux tmpfs; and I don't know if it would be possible to populate the tmpfs without a USB mass storage driver...

But once a mass storage driver is available, Redox could properly boot off of a read/write USB drive, without loading an FS into RAM.

**Comment by jackpot51** [comment]
I am working on USB, so it is likely we will have a way of having read/write root filesystems on USB sticks

**Comment by Pharaone** [comment]
I tried it with the new version, but the scripts stop here and the pc is stuck in the way
![e92221fa-e444-4760-a7ac-2e7821ad0d53](https://user-images.githubusercontent.com/17162630/29558227-c84d45ea-872b-11e7-8724-a0378c5a7522.jpeg)

**Comment by Pharaone** [comment]
still not working in 0.3.3

**Comment by pharaone** [comment]
Retried with 0.3.4, now is in a reboot loop after "gzip image ..."

**Comment by pharaone** [comment]
how can i get it working?

**Comment by MGageMorgan** [comment]
![20171117_151006_hdr](https://user-images.githubusercontent.com/13326489/32966912-6ddd3ef4-cba9-11e7-8d49-49a610303971.jpg)

This is a bit irritating...wasn't the goal of Rust efficiency? Correct me if I'm wrong, but shouldn't this take LESS memory than Linux?? I'm open to explanations, because I don't know WHY this is the way it is. Please enlighten me.

**Comment by jackpot51** [comment]
What exactly are you talking about @MGageMorgan ?

**Comment by MGageMorgan** [comment]
Look at the pic I posted. It was on a laptop with 2GB of RAM. That's standard. Why will it not run, and will it in the future?




Cheers!




M. Gage Morgan


----


Christoffen OSWorks LLC co-founder, CEO, and lead architect. 




/**


* Just note that my opinion will 


* NEVER reflect my company, which I 


* hope to keep neutral in terms of 


* public relations with other entities.


**/







On Fri, Nov 17, 2017 at 3:30 PM -0500, "Jeremy Soller" <notifications@github.com> wrote:












What exactly are you talking about @MGageMorgan ?

**Comment by jackpot51** [comment]
It is asking you to select a resolution. Press y to select or n to go to the next resolution

**Comment by MGageMorgan** [comment]
![15109569765021874220522](https://user-images.githubusercontent.com/13326489/32971401-171f4cf8-cbbb-11e7-9f9d-2ed3d56c13d8.jpg)

@jackpot51,  I understand that now (thanks), but regardless of which one I pick, it hangs here. Tried F1-4 like above, but can't get squat past this.

**Comment by jackpot51** [comment]
Thanks @MGageMorgan for testing. What computer is that? It would be good for debugging to know.

**Comment by MGageMorgan** [comment]
System76 Galago Pro. June 2017 build. Antergos atm. Intel i7 U-series piece-of-s***. 2GB RAM (couldn't afford more).

**Comment by jackpot51** [comment]
Where did you buy that from? System76 has only sold the Galago Pro with 4 GB to 16 GB of memory.

**Comment by MGageMorgan** [comment]
Nope. When I bought it from system76, they put TWO GIGS. I don't know why, at least thats what it said, let me double check, it might just me being a dumbass, but either way it doesn't work.

**Comment by MGageMorgan** [comment]
![screenshot from 2017-11-17 18-41-43](https://user-images.githubusercontent.com/13326489/32973701-1a499242-cbc7-11e7-8956-ded487961cba.png)

Confirmed: I'm a dumbass. But I don't think this exactly changes the problem at hand.

**Comment by jackpot51** [comment]
Thankfully I have access to the exact hardware you have so I will be able to look into it.

**Comment by MGageMorgan** [comment]
@jackpot51,

Let me know when you figure it out. I think the fact that you can dethrone C is amazing.

**Comment by MGageMorgan** [comment]
For the record, I'm tryna compile Redox from source on the same laptop. Currently jumping through the hoops that are PKGBUILDS to set up the Redox toolchain - I suspect the reason why Redox wouldn't build was because I didn't try to install them

**Comment by jackpot51** [comment]
Yep, the cross toolchain needs to be installed

**Comment by MGageMorgan** [comment]
@jackpot51 Did you get the thing to work on Galago or is it still broken? For some reason I still go out of my way to compile stuff I know won't work on my machine - I did it with ChromiumOS for a 32bit dell, now (possibly) Redox on a machine that doesn't like the prebuilt version. I don't know what's wrong with me lol

**Comment by jackpot51** [comment]
No, and it will probably be quite some time. In the meantime, running Redox in VirtualBox or QEMU should work.

**Comment by SirCmpwn** [comment]
I also get a boot loop attempting to run the latest Redox ISO off of a USB drive on a ThinkPad X200. This laptop has successfully run Linux, OpenBSD, NetBSD, Haiku, Minix, and plan9 before, so I assume the hardware is rather well understood.

**Comment by MGageMorgan** [comment]
@jackpot51 I DID NOT FUCKING REALIZE YOU WORKED AT SYSTEM76. Holy hell that must be a fun job...
